### PR TITLE
Difference between var and let

### PR DESCRIPTION
**Pull request by richzw**
## TL;DR;

In this tip, here are we talked about
-  The difference between `var` and `let`
-  Should we replace `let` with `var`?
-  use `let` safely.
## Username

 [@richzw](https://github.com/richzw)
## Extra

[content here]

### CODE REVIEW

File: _posts/en/2016-04-22-keyword-var-vs-let.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+          console.log(i); // output 0, 1, 2, 3, 4 
+      }, 100);  
+  }
+  ```
+  
+  More explaination could be found [here](http://stackoverflow.com/questions/30899612/explanation-of-let-and-block-scoping-with-for-loops/30900289#30900289).
+
+  [Here](https://jsbin.com/yumaye/edit?js,console) is the live demo of the above codes
+
+### Should I replace `var` with `let`?
```
Comment (loverajoel) [inline _posts/en/2016-04-22-keyword-var-vs-let.md:87]: I'm not sure about that

Comment (richzw) [inline _posts/en/2016-04-22-keyword-var-vs-let.md:87]: @loverajoel, you mean those codes do NOT work as expected?

``` js
  for (let i = 0; i < 5; ++i) {
      setTimeout(function () {
          console.log(i); // output 0, 1, 2, 3, 4 
      }, 100);  
  }
```

Comment (zenopopovici) [inline _posts/en/2016-04-22-keyword-var-vs-let.md:87]: Yep. https://bugs.chromium.org/p/v8/issues/detail?id=2560.

It's not wrong, it should be OK tough.

**Comment by loverajoel** [comment]
@richzw  can you add a playground like this https://github.com/loverajoel/jstips/blob/gh-pages/_posts/en/2016-08-10-comma-operaton-in-js.md ?

File: _posts/en/2016-04-22-keyword-var-vs-let.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+  for (let i = 0; i < 5; ++i) {
+      setTimeout(function () {
+          console.log(i); // output 0, 1, 2, 3, 4 
+      }, 100);  
+  }
+  ```
+  
+  More explaination could be found [here](http://stackoverflow.com/questions/30899612/explanation-of-let-and-block-scoping-with-for-loops/30900289#30900289).
+
+  [Here](https://jsbin.com/yumaye/edit?js,console) is the live demo of the above codes
```
Comment (loverajoel) [inline _posts/en/2016-04-22-keyword-var-vs-let.md]: is this your jsbin? can we use this as a playground?

Comment (richzw) [reply _posts/en/2016-04-22-keyword-var-vs-let.md]: @loverajoel, add playground per your request

**Comment by loverajoel** [comment]
@richzw thanks!

**Pull request merged**